src/Model/Subscription.php

```diff
@@ -210,13 +210,23 @@ public function getStartDate(): ?DateTime
     /**
      * @return Subscription
      */
-    public function setStartDate(?DateTime $startDate): self
+    public function setStartDate(DateTime $startDate): self
     {
         $this->startDate = $startDate;
 
         return $this;
     }
 
+    /**
+     * @return Subscription
+     */
+    public function beginImmediately()
```

Comment:
you did it! I take back what I said earlier. please forgive.

Reply:
circling back to this thought... what if we always took a start date... defaulted to today if one is not supplied...

then before we send off to BT... if the start date is today, we remove sending it to BT?

I see that bringing value to the end user by way of...
1. There is no confusion of a difference between a future start date, a start date of today, and no start date
1. The start date is ALWAYS going to be the date the subscription starts and you're always going to supply it.
1. No need to say begins immediately.
- and we're smart enough to know if it's today to just remove it before sending to BT

Reply:
Aren't null and today equivalent in this case? It might be easier to just default it to `null`. As soon as it is created, BT will return a valid `DateTime` indicating today right?

Reply:
There are some benefits to having to supply things...., but I think no date meaning right now seems pretty logical. Let's pair through this - maybe I am missing something.

Reply:
With a null date, `beginImmediately` should only need to be called when you are explicitly overriding a date.